### PR TITLE
handle asdot format and AS env variable

### PR DESCRIPTION
**Pull request by ishidawataru**
Signed-off-by: Wataru Ishida <ishida.wataru@lab.ntt.co.jp>

### CODE REVIEW

**Comment by robbrockbank** [comment]
Hi @ishidawataru,

Apologies that we have been sitting on your PRs!  We're just in the middle of releasing a new RC so we're not introducing any new function at this point, but we should be in a good position to review these and get them merged in soon!   It will be great to have feature parity with BIRD - so definitely something we're keen to get reviewed!

Rob

**Comment by ishidawataru** [comment]
@robbrockbank Thanks. no problem :) Sorry for opening PRs in busy time!

File: main.go [hunk 1]
```diff
@@ -91,29 +93,25 @@ func getEtcdConfig() (etcd.Config, error) {
...
-	if err != nil {
-		return 0, err
-	}
-	asn, err := strconv.ParseUint(res.Node.Value, 10, 32)
-	if err != nil {
-		return 0, err
+func getGlobalASN(api etcd.KeysAPI) (numorstring.ASNumber, error) {
+	value := os.Getenv(AS)
+	if value == "" {
+		res, err := api.Get(context.Background(), fmt.Sprintf("%s/global/as_num", CALICO_BGP), nil)
```
Comment (gunjan5) [inline main.go]: you might be able to use libcalico's `GetGlobalASNumber()` here.
https://github.com/projectcalico/libcalico-go/blob/master/lib/client/config.go#L59

File: main.go [hunk 2]
```diff
@@ -91,29 +93,25 @@ func getEtcdConfig() (etcd.Config, error) {
...
+			return 0, err
+		}
+		value = res.Node.Value
 	}
-	return uint32(asn), nil
+	return numorstring.ASNumberFromString(value)
 }
 
-func getPeerASN(api etcd.KeysAPI, host string) (uint32, error) {
+func getPeerASN(api etcd.KeysAPI, host string) (numorstring.ASNumber, error) {
```
Comment (gunjan5) [inline main.go]: you should be able to use libcalico `Node` resource here to get the host ASN.
see: https://github.com/projectcalico/libcalico-go/blob/master/lib/api/node.go#L62

**Comment by gunjan5** [comment]
2 places where libcalico API could be used so that this can also work with non-etcd backends

**Comment by ishidawataru** [comment]
@gunjan5 Oops. closed mistakenly. Thanks for the review.
I'll reflect your comments shortly.

**Comment by gunjan5** [comment]
no worries, seems like it was from your fork's `master` branch anyway. You might find it useful to keep master just to sync your fork with the upstream

**Pull request closed without merging**